js_modules/dagit/packages/core/src/overview/OverviewAssetsRoot.tsx

```diff
@@ -0,0 +1,433 @@
+      const assets = query.data.assetsOrError.nodes;
+      return {assets, groupedAssets: groupAssets(assets)};
+    }
+    return {assets: [], groupedAssets: []};
+  }, [query.data?.assetsOrError]);
+
+  const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: groupedAssets.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 82,
+    overscan: 5,
+  });
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  function content() {
+    const result = query.data?.assetsOrError;
+    if (!query.data && query.loading) {
+      return (
+        <Box
+          flex={{alignItems: 'center', justifyContent: 'center', direction: 'column', grow: 1}}
+          style={{width: '100%'}}
+        >
+          <Spinner purpose="page" />
+        </Box>
+      );
+    }
+    if (result?.__typename === 'PythonError') {
+      return (
+        <Box
+          flex={{alignItems: 'center', justifyContent: 'center', direction: 'column', grow: 1}}
+          style={{width: '100%'}}
+        >
+          <PythonErrorInfo error={result} />
+        </Box>
+      );
+    }
+
+    return (
+      <Box flex={{direction: 'column'}} style={{overflow: 'hidden'}}>
+        <Container ref={parentRef}>
+          <VirtualHeaderRow />
+          <Inner $totalHeight={totalHeight}>
+            {items.map(({index, key, size, start}) => {
+              const group = groupedAssets[index];
+              return <VirtualRow key={key} start={start} height={size} group={group} />;
+            })}
+          </Inner>
+        </Container>
+      </Box>
+    );
+  }
+
+  const [searchGroup, setSearchGroup] = useQueryPersistedState<AssetGroupSelector | null>({
+    queryKey: 'g',
+    decode: (qs) => (qs.group ? JSON.parse(qs.group) : null),
+    encode: (group) => ({group: group ? JSON.stringify(group) : undefined}),
+  });
+
+  return (
+    <>
+      <div style={{position: 'sticky', top: 0, zIndex: 1}}>
+        <Header refreshState={refreshState} />
+        <Box
+          padding={{horizontal: 24, vertical: 16}}
+          flex={{alignItems: 'center', gap: 12, grow: 0}}
+        >
+          <TabButton selected="assets" />
+          <AssetGroupSuggest assets={assets} value={searchGroup} onChange={setSearchGroup} />
+        </Box>
+      </div>
+      {content()}
+    </>
+  );
+};
+
+type Assets = Extract<
+  AssetCatalogTableQuery['assetsOrError'],
+  {__typename: 'AssetConnection'}
+>['nodes'];
+
+function groupAssets(assets: Assets) {
+  const groups: Record<
+    string,
+    {
+      groupName: string;
+      repositoryName: string;
+      assets: Assets;
+    }
+  > = {};
+
+  assets.forEach((asset) => {
+    if (!asset.definition) {
+      return;
+    }
+    const groupName = asset.definition.groupName;
+    const repositoryName = asset.definition.repository.name;
```

Comment:
If the repository location value is also available, it might be wise to use a full humanized RepoAddress string here -- otherwise you might end up with repo name collisions.